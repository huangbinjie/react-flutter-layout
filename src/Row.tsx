import React from 'react'
import { Overflow } from './style'

export interface IRow {
  className?: string
  overflow?: Overflow
}

/**
 * 上下布局的组件。
 * 高宽默认占满父容器。
 * 默认占满父容器
 */
export default function Row(props: React.PropsWithChildren<IRow>) {
  const {
    children,
    overflow = 'hidden',
    ...restProps
  } = props

  const style: React.CSSProperties = {
    display: 'flex',
    flexDirection: 'row',
    overflow
  }

  return <div style={style} {...restProps}>{children}</div>
}
