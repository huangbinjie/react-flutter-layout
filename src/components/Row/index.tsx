import React from 'react'
import { style as jss } from 'typestyle'
import classnames from 'classnames'
import {
  AxisSize,
  DirectionContext,
  Alignment,
  decodeAlignment,
  decodeAxisSize
} from '../../style'

export type RowProps = {
  className?: string
  mainAxisAlignment?: Alignment
  crossAxisAlignment?: Alignment
  mainAxisSize?: AxisSize
  crossAxisSize?: AxisSize
  style?: React.CSSProperties
}

/**
 * 上下布局的组件。
 * 高度是最高的子组件的高度。
 * 宽度默认最大
 * 高度默认最小
 */
export default function Row({
  className = '',
  crossAxisAlignment = 'normal',
  mainAxisAlignment = 'start',
  mainAxisSize = 'max',
  crossAxisSize = 'min',
  children,
  ...restProps
}: React.PropsWithChildren<RowProps>) {
  const classname = jss({
    display: 'flex',
    flexDirection: 'row',
    alignItems: decodeAlignment(crossAxisAlignment),
    justifyContent: decodeAlignment(mainAxisAlignment),
    width: decodeAxisSize(mainAxisSize),
    height: decodeAxisSize(crossAxisSize)
  })

  return (
    <DirectionContext.Provider value="row">
      <div
        debug-label="Row"
        className={classnames(classname, className)}
        {...restProps}
      >
        {children}
      </div>
    </DirectionContext.Provider>
  )
}
