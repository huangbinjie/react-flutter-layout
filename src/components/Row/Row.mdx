---
name: Row
menu: components
---

import { Playground, Props } from 'docz'
import Container from '../Container'
import Row from '../Row'
import Center from '../Center'

## Row

提供水平布局的组件。等价于 `flex-direction: horizontal`

### Basic

<Playground>
  <Row>
    <Container width="100px" height="100px" color="#ececec">
      <Center>居中</Center>
    </Container>
    <Container width="100px" height="100px" color="blue">
      <Center style={{color: "#fff"}}>居中</Center>
    </Container>
  </Row>
</Playground>

## Props

<Props of={Row} />
