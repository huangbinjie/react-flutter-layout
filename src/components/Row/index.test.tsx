import React from 'react'
import renderer from 'react-test-renderer'

import Row from '.'

describe('Row', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Row>empty content</Row>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
