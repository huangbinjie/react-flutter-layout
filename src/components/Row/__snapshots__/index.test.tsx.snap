// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row basic 1`] = `
<div
  className="foxw8m9"
  debug-label="Row"
>
  empty content
</div>
`;
